sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala

```diff
@@ -631,18 +631,20 @@ class ExecChecks private(
     val plan = meta.wrapped.asInstanceOf[SparkPlan]
     val allowDecimal = meta.conf.decimalTypeEnabled
 
-    val unsupportedOutputTypes = plan.output
-      .filterNot(attr => check.isSupportedByPlugin(attr.dataType, allowDecimal))
-      .toSet
+    val unsupportedOutputTypes = plan.output.collect {
+      case attr if !check.isSupportedByPlugin(attr.dataType, allowDecimal) => attr.dataType
```

Comment:
This was an unfortunate regression introduced by https://github.com/NVIDIA/spark-rapids/commit/64f9d9b4414103fa424283712371e2b9bddcd8fe#diff-c5cb89e178628729e961e2e0e9be2689893254638c5ce467250a263f7867112dR558